toolbox/metrics/flakes/flakes.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flakes
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+
+	u "istio.io/test-infra/toolbox/util"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	bucket             = "k8s-metrics"
+	latestFlakesMetric = "istio-job-flakes-latest.json"
```

Comment:
Just for information, who is creating this? Is there a way to get flakiness per org/repo/branch ?

Reply:
The PR linked in this PR description adds a new metric config to https://github.com/kubernetes/test-infra/tree/master/metrics/configs and k8s infra team has already set up a job to build all metrics every 24 hours and upload results to `gs://k8s-metrics`. That PR is still pending, and for the purpose of an example, an output that I created is at `gs://istio-flakes/istio-job-flakes-latest.json`.

I think it is possible to get this metric per repo and/or per branch, as such information is probably tracked in the [big query instance](https://bigquery.cloud.google.com/table/k8s-gubernator:build.all?pli=1). However, this metric that I added is meant for just istio because of the way we use Prow is different from k8s. Existing flakiness metric config contains SQL query that depends on differences between job names for presubmis and postsubmits (`ci-`/`pr-` prefix) and the `version` field must not be "unknown", neither is true for istio. So doing this per org is out of the question in this case.

Reply:
this might get resolved when we switch to podutils.